### PR TITLE
Adding SDLF Lightweight Example

### PR DESCRIPTION
**Pull request by anmolsgandhi**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by anmolsgandhi** [comment]
Important file to review:

sdlf-ddk-lightweight/data_lake/pipelines/sdlf_pipeline_stack.py
sdlf-ddk-lightweight/data_lake/pipelines/parameters.json
sdlf-ddk-lightweight/data_lake/stages/foundations_stage.py
sdlf-ddk-lightweight/data_lake/stages/sdlf_dataset_stage.py
sdlf-ddk-lightweight/data_lake/stages/sdlf_heavy_transform.py
sdlf-ddk-lightweight/data_lake/stages/sdlf_light_transform.py
sdlf-ddk-lightweight/app.py
sdlf-ddk-lightweight/scripts/cleanup_scripts/
sdlf-ddk-lightweight/README.md

File: sdlf-ddk-lightweight/app.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from typing import Any, Dict
+import aws_cdk as cdk
+from aws_ddk_core.cicd import CICDPipelineStack
+#from utils.application_config import GetApplicationParameters
```
Comment (malachi-constant) [inline sdlf-ddk-lightweight/app.py:19]: Is this leftover from something? Or do we plan to use that method at some point?

File: sdlf-ddk-lightweight/data_lake/pipelines/sdlf_pipeline_stack.py [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+                stage_bucket_key = self._foundations_stage.stage_bucket_key,
+                data_lake_lib = self._foundations_stage.data_lake_library,
+                register_provider = self._foundations_stage.register_provider,
+                wrangler_layer = self._wrangler_layer,
+                database_crawler_name = f"{resource_prefix}-{team}-{dataset}-post-stage-crawler"
+            ),
+        )
+        self.routing_b = data_lake_heavy_transform.routing_lambda
+    
+        self._data_lake_pipeline: DataPipeline = (
```
Comment (malachi-constant) [inline sdlf-ddk-lightweight/data_lake/pipelines/sdlf_pipeline_stack.py:205]: Nice!

File: sdlf-ddk-lightweight/data_lake/pipelines/sdlf_pipeline_stack.py [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+            .add_stage(data_lake_s3_event_capture_stage)
+            .add_stage(data_lake_light_transform)
+            .add_stage(data_lake_heavy_transform, skip_rule=True)
+        )
+        return data_lake_heavy_transform
+
+    def _create_custom_pipeline(self):
+        return
+
+    def _create_wrangler_layer(self):
```
Comment (malachi-constant) [inline sdlf-ddk-lightweight/data_lake/pipelines/sdlf_pipeline_stack.py:221]: Are we assuming this example only works in `us-east-1`?

File: sdlf-ddk-lightweight/README.md [hunk 4]
```diff
@@ -0,0 +1,458 @@
...
+- [Deployment](#deployment)
+- [Parameters](#parameters)
+- [Resources Deployed](#resources-deployed)
+
+<br />
+<br />
+
+## Reference Architecture
+
+![Alt](/docs/images/SDLF-Data-Lake-DDK.png)
```
[the remark this answers is not in the document]
Comment (anmolsgandhi) [reply sdlf-ddk-lightweight/README.md:27]: will update, thanks

File: sdlf-ddk-lightweight/requirements.txt [hunk 5]
```diff
@@ -0,0 +1,3 @@
+aws_ddk_core==0.4.0
```
[the remark this answers is not in the document]
Comment (anmolsgandhi) [reply sdlf-ddk-lightweight/requirements.txt:1]: No specific reason, i think we might have missed it. should we update it to 0.5.1 ?

File: sdlf-ddk-lightweight/app.py (hunk 1 above)
Comment (anmolsgandhi) [reply sdlf-ddk-lightweight/app.py:19]: Will remove, no longer needed

File: sdlf-ddk-lightweight/data_lake/pipelines/sdlf_pipeline_stack.py (hunk 3 above)
Comment (anmolsgandhi) [reply sdlf-ddk-lightweight/data_lake/pipelines/sdlf_pipeline_stack.py:221]: It should be region agnostic, but i am not sure which regions support aws-data-wrangler lambda layer. i can parameterize region

**Comment by anmolsgandhi** [comment]
Updated PR with new commit resolving comments

File: sdlf-ddk-lightweight/data_lake/pipelines/sdlf_pipeline_stack.py (hunk 3 above)
Comment (malachi-constant) [reply sdlf-ddk-lightweight/data_lake/pipelines/sdlf_pipeline_stack.py:221]: We support all commercial regions now for wrangler layers.

**Review by malachi-constant — APPROVED** [review]

File: sdlf-ddk-lightweight/ddk.json [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    "dev": {
+      "account": "111111111111",
+      "region": "us-east-1",
+      "name": "Development",
+      "manual_approvals": false,
+      "email": "nobody@demo.com",
+      "execute_security_lint": true,
+      "execute_tests": false,
+      "resource_prefix": "sdlf",
+      "data_pipeline_parameters": {
```
Comment (jaidisido) [inline sdlf-ddk-lightweight/ddk.json:15]: I would think these parameters would be global, not environment specific?

**Review by jaidisido — APPROVED** [review]
Some minor comments - but overall great work!

File: sdlf-ddk-lightweight/data_lake/stages/foundations_stage.py [hunk 7]
```diff
@@ -0,0 +1,538 @@
...
+from aws_cdk.aws_kms import Key, IKey
+from aws_cdk.aws_lakeformation import CfnResource
+from aws_cdk.aws_lambda import Code, Function, Runtime, LayerVersion, IFunction, ILayerVersion
+from aws_cdk.aws_s3 import Bucket, BucketEncryption, BlockPublicAccess, BucketAccessControl, IBucket
+from aws_cdk.aws_ssm import StringParameter
+from aws_cdk.custom_resources import Provider
+from aws_cdk.aws_s3_deployment import BucketDeployment, ServerSideEncryption, Source
+from aws_ddk_core.resources import S3Factory, KMSFactory, LambdaFactory
+
+class FoundationsStage(DataStage):
```
Comment (kukushking) [inline sdlf-ddk-lightweight/data_lake/stages/foundations_stage.py:33]: Could you explain what is the reason for making `FoundationsStage` a `DataStage`? It does not seem to produce an event or be chained with any other data transformation or ingestion stage.

Could this be a stack inheriting from `BaseStack` instead?

File: sdlf-ddk-lightweight/data_lake/stages/sdlf_dataset_stage.py [hunk 8]
```diff
@@ -0,0 +1,236 @@
...
+    stage_a_transform: str
+    stage_b_transform: str
+    artifacts_bucket: IBucket
+    artifacts_bucket_key: IKey
+    stage_bucket: IBucket
+    stage_bucket_key: IKey
+    glue_role: IRole
+    register_provider: Provider
+
+class SDLFDatasetStage(DataStage):
```
Comment (kukushking) [inline sdlf-ddk-lightweight/data_lake/stages/sdlf_dataset_stage.py:46]: Same as above

**Review by kukushking — COMMENTED** [review]
Very cool to finally have an E2E SDLF example on DDK - and amazing work on light/heavy transform stages & data lake library with multi-dataset routing - it is something that we wanted to have for a long time! Left a couple of logical comments but overall impressive work! 👏

File: sdlf-ddk-lightweight/data_lake/stages/sdlf_light_transform.py [hunk 9]
```diff
@@ -0,0 +1,678 @@
...
+            environment={
+                "TEAM": self.team,
+                "PIPELINE": self.pipeline,
+                "STAGE": "StageA"
+            },
+            role = redrive_lambda_role,
+            description="Redrive Step Function stageA",
+            timeout=cdk.Duration.minutes(10),
+            memory_size=256,
+            runtime = Runtime.PYTHON_3_9,
```
Comment (LeonLuttenberger) [inline sdlf-ddk-lightweight/data_lake/stages/sdlf_light_transform.py:550]: I wonder if it makes sense to add the runtime version as a parameter for the stage. That way, when Python 3.9 becomes outdated, the customer only needs to change the runtime in one place (per transformation pipeline at least).

**Review by LeonLuttenberger — COMMENTED** [review]
I left a couple of minor nitpicky comments but overall great work!

File: sdlf-ddk-lightweight/LICENSE.txt [hunk 10]
```diff
@@ -0,0 +1,202 @@
+                                 Apache License
```
[the remark this answers is not in the document]
Comment (noah-paige) [reply sdlf-ddk-lightweight/LICENSE.txt:1]: Sounds good - will remove

**Review by kukushking — APPROVED** [review]

**Pull request merged**